From 8a7d6b732a74ccbff1ced447b31b297d53560e44 Mon Sep 17 00:00:00 2001
From: Jonas Larsson <ljonas@google.com>
Date: Tue, 30 Oct 2018 14:36:06 -0700
Subject: [PATCH 1/6] kmssink: fix bad property memory management

When the driver-name property is read it incorrectly
transfers ownership of the underlying string to the
reader of the property, causing crashes.

This prevented using GST_DEBUG_DUMP_DOT_DIR to dump
pipeline graphs containing kmssink, as the dumping
process always reads all properties.

Change-Id: I5432ca1e3eca793153fdab93909a39bbc1698720
---
 sys/kms/gstkmssink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index 453b00cc3..e38d638f8 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -2172,7 +2172,7 @@ gst_kms_sink_get_property (GObject * object, guint prop_id,
 
   switch (prop_id) {
     case PROP_DRIVER_NAME:
-      g_value_take_string (value, sink->devname);
+      g_value_set_string (value, sink->devname);
       break;
     case PROP_BUS_ID:
       g_value_take_string (value, sink->bus_id);
-- 
2.26.0

